_posts/2019-10-23-kafka-authentication-using-oauth-2.0.md

```diff
@@ -0,0 +1,538 @@
+Each `client` has to be configured on the server.
+
+We'll manually configure a `client` for Kafka broker - we'll simply call it 'kafka-broker'.
+
+With the `master` realm open, click on `Clients` in the main navigation on the left.
+Now use the `Create` button at the right top corner.
+In the form that opens set `Client ID` to `kafka-broker` and click `Save`.
+The `Settings` view opens up where you change the `Access Type` to `Confidential`.
+Then turn `Standard Flow Enabled` to `OFF`, and `Service Accounts Enabled` to `ON`.
+Also, scroll down to `Advanced Settings` section, and set `Access Token Lifespan` to 10 minutes.
+The reason for this is because, by default, access tokens are very short-lived as they are configured for use by browser-based UI flows.
+For Kafka authentication we should make their expiry time slightly longer to get rid of the warnings that result from Kafka client library refresh defaults.
+Now, click `Save`.
+Switch to another tab that appears, called `Credentials`.
+Leave `Client Authenticator` set to `Client Id and Secret`, and take note of the `Secret`. We'll need it later.
+
+What we've done here is configure the kafka-broker 'client' in a way that it can directly authenticate with Keycloak in its own name, using its own Client ID and Secret, and at the same time can't be used for browser based login by other users.
+This is a typical 'client' configuration for microservices, regardless of whether they are acting as application clients, or application servers. 
+
+![Create Client in Keycloak Console](/assets/2019-10-23-keycloak.png)
+
+All Kafka brokers will share the same 'client' configuration since they are just different instances of the same service.
+
+Next, we configure a 'client' for each service that acts as a Kafka client. These are Kafka producers, consumers, connectors, streams applications - any application that talks to Kafka brokers. 
+
+In general this is done by following the same steps as for 'kafka-broker', but giving each new client its own `Client ID`, and taking note of the generated `Secret`.
+
+For this demo we'll need two more clients: 'kafka-producer' and 'kafka-consumer'.
+
+When you are done it is wise to export the current state of the realm configuration.
+Since we are running Keycloak as a stateless pod, it will happen sooner rather than later that the pod will be deleted and replaced with a fresh new one, and all state will be gone.
+Exporting the realm gives us a way to restore this state when this happens.
+
+To export the realm click on `Export` in main navigation on the left.
+
+Then in `Partial Export` page turn `Export clients` to ON, and click `Export`.
+Note, that secrets in the exported file get lost, and if you ever perform `Import`, you'll have to use the `Regenerate Secret` button in the `Credentials` tab of each client.
+
+
+
+## Installing Strimzi
+
+Before we can start a Kafka cluster we first have to install the custom resource definitions (CRDs) for Strimzi.
```

Comment:
The step below actually installs whole Strimzi, not just the CRDs.

Reply:
What about:
'Before we can start a Kafka cluster we first have to install the custom resource definitions (CRDs) and the Kubernetes operators provided by Strimzi.'